### ISSUE TITLE
Frames drop when adding several models - Geospatial

### ISSUE THREAD

**Issue by Jose-Develaw** [SceneView/sceneview-android#260]
I am experimenting a huge (I mean huge) frame dropping when adding several 3D models to the scene. I am using Geospatial API and I don't know what I am doing wrong.

Previously I use an OpenGL implementation and I had not those kind of problems, but I change to SceneView cause of the easy tap management and the Compose style.

Can anyone show me a better way for doing this? One that can improve the performance. I really need to be able of showing several models.

Note: MyArModelNode inherits ArModelNode and is used for store some additional properties, so I can get them back on tap.
Note: the GLB is 2.7MB... is that too much? I have no trouble with it in iOS with ARKit.

Thanks in advance.

My code:

```
onFrame = { arFrame ->
                    if (nodes.isEmpty()) {

                        if (!arFrame.session.earth?.anchors?.isEmpty()!!) {
                            for (anchor in arFrame.session.earth?.anchors!!) {
                                anchor.detach()
                            }
                        }
                        messages.map { message ->
                        
                        //Here I am limiting the models I load
                            if (nodes.count() < 10) {

                                arFrame.session.earth.let { earth ->
                                    if (earth?.trackingState == TrackingState.TRACKING) {
                                        // Place the earth anchor at the same altitude as that of the camera to make it easier to view.
                                        val altitude = earth.cameraGeospatialPose.altitude - 0.8
                                        val rotation = Rotation(0f, 0f, 0f)
                                        // Put the anchor somewhere around the user.
                                        val latitude = message.latitude
                                        val longitude = message.longitude
                                        val earthAnchor = earth.resolveAnchorOnTerrain(
                                            latitude, longitude,
                                            0.0, 0f, 0f, 0f, 1f
                                        )

                                        val arNode = MyArModelNode(
                                        ).apply {
                                            scaleModel(10.0f)
                                            isVisible = true
                                            loadModelGlbAsync(glbFileLocation = "models/stick_voizeland.glb")
                                        }

                                        arNode.anchor = earthAnchor
                                        arNode.messageId = message.id
                                        addChild(arNode)
                                        nodes.add(arNode)
                                    }

                                }
                            }
                        }
                    }
            }
